webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/reassignHostGroup/BulkReassignHostgroupModal.js

```diff
@@ -0,0 +1,194 @@
+        search: fetchBulkParams(),
+      },
+      hostgroup_id: hostgroupId,
+    };
+
+    dispatch(
+      bulkReassignHostgroups(
+        requestBody,
+        () => visit(foremanUrl('/new/hosts')),
+        handleError
+      )
+    );
+  };
+
+  const modalActions = [
+    <Button
+      key="add"
+      ouiaId="bulk-reassign-hg-modal-add-button"
+      variant="primary"
+      onClick={handleSave}
+      isDisabled={hostUpdateStatus === STATUS.PENDING}
+      isLoading={hostUpdateStatus === STATUS.PENDING}
+    >
+      {__('Save')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-reassign-hg-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Change host group')}
+      width="50%"
+      position="top"
+      actions={modalActions}
+      id="bulk-reassign-hg-modal"
+      key="bulk-reassign-hg-modal"
+      ouiaId="bulk-reassign-hg-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-reassign-hg-options">
+          <FormattedMessage
+            defaultMessage={__(
+              'Change the host group of {hosts}. Some hosts may already be in your chosen host group.'
+            )}
+            values={{
+              hosts: (
+                <strong>
+                  <FormattedMessage
+                    defaultMessage="{count, plural, one {# {singular}} other {# {plural}}}"
+                    values={{
+                      count: selectedCount,
+                      singular: __('selected host'),
+                      plural: __('selected hosts'),
+                    }}
+                    id="bulk-hg-selected-host-options"
+                  />
+                </strong>
+              ),
+            }}
+            id="bulk-reassign-hg-description"
+          />
+        </Text>
+        {hostgroups && hostgroupStatus === STATUS.RESOLVED && (
+          <>
+            <FormGroup type="string" fieldId="host_group">
+              <FormSelect
+                ouiaId="hostgroup-select"
+                isRequired
+                value={hostgroupId}
+                onChange={value => {
+                  setHostgroupId(value);
+                }}
+                aria-label="hostgroup_select"
+              >
+                {[
+                  <FormSelectOption
+                    key="placeholder"
+                    value=""
+                    label={__('Select a hostgroup')}
+                    isDisabled
+                  />,
+                ].concat(
+                  hostgroups?.results?.map(hg => (
+                    <FormSelectOption
+                      key={hg.id}
+                      value={hg.id}
+                      label={hg.name}
+                    />
+                  ))
+                )}
+              </FormSelect>
```

Comment:
Lots of customers have hundreds of hostgroups. How about a `Select` with `variant={SelectVariant.typeahead}` like we do in `<ContentViewSelect>`? (Check out the 'Change content view environments' action for an example) 

This way you could type the name of a hostgroup and not have to scroll thru the list.